src/account/utils/Types.ts

```diff
@@ -178,26 +178,28 @@ export type BiconomySmartAccountV2ConfigConstructorProps =
     ResolvedBundlerProps &
     ResolvedValidationProps
 
+/**
+ * Represents options for building a user operation.
+ * @typedef BuildUserOpOptions
+ * @property {GasOffset} [gasOffset] - Increment gas values by giving an offset, the given value will be an increment to the current estimated gas values, not an override.
+ * @property {ModuleInfo} [params] - Parameters relevant to the module, mostly relevant to sessions.
+ * @property {NonceOptions} [nonceOptions] - Options for overriding the nonce.
+ * @property {boolean} [forceEncodeForBatch] - Whether to encode the user operation for batch.
+ * @property {PaymasterUserOperationDto} [paymasterServiceData] - Options specific to transactions that involve a paymaster.
+ * @property {SimulationType} [simulationType] - Determine which parts of the transaction a bundler will simulate: "validation" | "validation_and_execution".
+ * @property {StateOverrideSet} [stateOverrideSet] - For overriding the state.
+ * @property {boolean} [useEmptyDeployCallData] - Set to true if the transaction is being used only to deploy the smart contract, so "0x" is set as the user operation call data.
+ */
 export type BuildUserOpOptions = {
-  /** overrides: Explicitly set gas values */
-  // overrides?: Overrides;
-  /** Not currently in use  */
-  // skipBundlerGasEstimation?: boolean;
-  /**  params relevant to the module, mostly relevant to sessions */
-  params?: ModuleInfo
-  /**  nonceOptions: For overriding the nonce */
-  nonceOptions?: NonceOptions
-  /**  forceEncodeForBatch: For encoding the user operation for batch */
-  forceEncodeForBatch?: boolean
-  /**  paymasterServiceData: Options specific to transactions that involve a paymaster */
-  paymasterServiceData?: PaymasterUserOperationDto
-  /**  simulationType: Determine which parts of the tx a bundler will simulate: "validation" | "validation_and_execution".  */
-  simulationType?: SimulationType
-  /**  stateOverrideSet: For overriding the state */
-  stateOverrideSet?: StateOverrideSet
-  /** set to true if the tx is being used *only* to deploy the smartContract, so "0x" is set as the userOp.callData  */
-  useEmptyDeployCallData?: boolean
-}
+  gasOffset?: GasOffset;
```

Comment:
Instead of it being an amount that you increment as an absolute value, could it not be expressed as a percentage? Like I would like to be able to increase the amount be 10% by providing an 'offsetFactor' of '+0.1', where the original estimate gets multiplied by `1.1`. I think this might be more useful

Reply:
Let me check how this would be implemented, if it adds too much complexity I don't think it's worth it

Reply:
We could even hardcode the offsetFactors: `1`, `1.2` & `1.4` and then offer them as "SLOW", "MEDIUM" and "FAST" for the sake of familiarity of MM users

Reply:
Interesting, like "slippage" in DeFi, will look into this

Reply:
"SLOW", "MEDIUM" and "FAST" doesn't make sense in this situation as these increments are not increasing tx speed, like on Metamask